Source/ORTS.Common/Filter.cs

```diff
@@ -281,19 +257,13 @@ public enum FilterTypes
         /// <returns>Filtered value</returns>
         public float Filter(float NewSample)
         {
-            //shift the old samples
-            for (int n = NCoef; n > 0; n--)
-            {
-                x[n] = x[n - 1];
-                y[n] = y[n - 1];
-            }
             //Calculate the new output
-            x[0] = NewSample;
-            y[0] = (float)Convert.ToDouble(ACoef[0]) * (float)Convert.ToDouble(x[0]);
+            x.Insert(0, NewSample);
+            y.Insert(0, ACoef[0] * x[0]);
```

Comment:
I don't understand this modification. You are adding new elements to the list, which will make it grow unlimited, while the previous implementation was just shifting values and discarding the oldest ones, so it had fixed size.

I'd suggest using plain arrays instead of lists. The filter will always have constant size (actually only NCoef=1 is implemented). Also, ComputeButterworth has a terrible implementation, with the list being cleared and rebuilt again. So this has to be refactored in a way such that the allocations only happen at the constructor.

Reply:
César, you are right in these! I've fixed my mistakes by your suggestion.